C/c4Database.cc

```diff
@@ -89,28 +89,55 @@ bool c4key_setPassword(C4EncryptionKey *key, C4String password, C4EncryptionAlgo
 C4Database* c4db_open(C4Slice path,
                       const C4DatabaseConfig *configP,
                       C4Error *outError) noexcept
+{
+    return dbOpen(path, configP, DatabaseTagExternal, outError);
+}
+
+// The corresponding internally-faced API
+C4Database* dbOpen(C4String path,
+                   const C4DatabaseConfig *configP,
+                   C4DatabaseTag dbTag,
```

Comment:
This seems like too much code change ... why not just add the C4DatabaseTag to the C4DatabaseConfig?

Reply:
In that meeting, c.f. note of CBL-2357, we decided not to change the public APIs.
The goal was to make sure that c4db_openXs() are only called by the platform. Internally we can only call dbOpenXs(). This way, we will know what DataFiles are related to the applications. This distinction has business semantics (fail-fast), whereas other tags only contribute to the error message.

Reply:
Also, if we make it a DatabaseConfig, the customer will be responsible to to fill up config correctly.